src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -1344,6 +1501,90 @@ ShardLength(uint64 shardId)
 }
 
 
+/*
+ * NodeGroupGetSeparatedShardPlacementGroup returns the shard placement group
+ * that given node group is used to separate from others. Returns NULL if this
+ * node is not used to separate a shard placement group.
+ */
+ShardPlacementGroup *
+NodeGroupGetSeparatedShardPlacementGroup(int32 groupId)
+{
+	ShardPlacementGroup *nodeShardPlacementGroup = NULL;
+	bool shardPlacementGroupNeedsSeparateNode = false;
+
+	bool indexOK = false;
+	int scanKeyCount = 1;
+	ScanKeyData scanKey[1];
+
+	Relation pgDistPlacement = table_open(DistPlacementRelationId(),
+										  AccessShareLock);
+
+	ScanKeyInit(&scanKey[0], Anum_pg_dist_placement_groupid,
+				BTEqualStrategyNumber, F_INT4EQ, Int32GetDatum(groupId));
+	SysScanDesc scanDescriptor = systable_beginscan(pgDistPlacement,
+													DistPlacementGroupidIndexId(),
+													indexOK,
```

Comment:
`indexOk` has been set to false, forcing a heap scan. Any reason to tell postgres the index is not ok to use at the moment?